source/core_mqtt_serializer.c

```diff
@@ -1164,8 +1164,15 @@ static MQTTStatus_t deserializeSuback( const MQTTPacketInfo_t * pSuback,
         LogDebug( ( "Packet identifier %hu.",
                     ( unsigned short ) *pPacketIdentifier ) );
 
-        status = readSubackStatus( remainingLength - sizeof( uint16_t ),
-                                   pVariableHeader + sizeof( uint16_t ) );
+        if( *pPacketIdentifier == 0U )
```

Comment:
Should the `MQTT_PACKET_ID_INVALID` macro be moved the serializer API?https://github.com/FreeRTOS/coreMQTT/blob/main/source/include/core_mqtt.h#L55-L61
```suggestion
        if( *pPacketIdentifier == MQTT_PACKET_ID_INVALID )
```

Reply:
Oh wow, I forgot about this. Possibly, but doing that is a larger change than what this PR was focused on.